packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -63,10 +65,15 @@ function addVNodeToChildLWC(vnode: VCustomElement) {
 }
 
 // [s]tatic [p]art
-function sp(partId: number, data: VStaticPartData): VStaticPart {
+function sp(partId: number, content: VStaticPartData & VStaticText): VStaticPart {
+    // Static part will always have either text or data, it's guaranteed by the compiler.
+    const { text, ...data } = content;
```

Comment:
We're doing an extra object allocation/clone here. Can we directly pass in the `text` separate from the data by the compiler?

Reply:
Updated in the latest commit - 4024f8f5a2a5e9d9ce977db815e989df0a97e559